packages/ember-debug/lib/error.js

```diff
@@ -18,15 +7,12 @@ function ExtendBuiltin(klass) {
   @constructor
   @public
 */
-export default class EmberError extends ExtendBuiltin(Error) {
-  constructor(message) {
-    super();
 
-    if (!(this instanceof EmberError)) {
-      return new EmberError(message);
-    }
+export default EmberError;
+function EmberError(message, code) {
+  if (this instanceof EmberError) {
+    let error = Error.call(this, message, code);
```

Comment:
FYI: In V8, using `super()` would give you the benefit of omitting all frames up to and including the `EmberError` constructor (which is what I believe `Error.captureStackTrace` was used for). ES6 `Error` subclasses are basically the better `Error.captureStackTrace`. Copying error properties below would also be unnecessary.

I'm not sure how other browsers behave though.